### PR TITLE
Backend file

### PR DESCRIPTION
**Pull request by jacobweinstock**
## Description

This adds a file watcher backend. Useful for local development and testing.

## Why is this needed
Helps with local development and testing.

Fixes: #

## Checklist:

I have:

- [X] updated the documentation and/or roadmap (if required)
- [X] added unit or e2e tests
- [X] provided instructions on how to upgrade

### CODE REVIEW

File: example/main.go [hunk 1]
```diff
@@ -0,0 +1,55 @@
+package main
```
Comment (displague) [inline example/main.go]: Let's move this to `examples/something/`

Comment (jacobweinstock) [reply example/main.go]: sure thing. updated.

**Comment by jacobweinstock** [comment]
Hey @nshalman, you are the only approver in this repo (I need to change this), would you mind checking this out?
@displague, pinging you for another review, please.

File: backend/file/file.go [hunk 2]
```diff
@@ -0,0 +1,253 @@
...
+		if strings.EqualFold(k, mac.String()) {
+			// found a record for this mac address
+			v.MACAddress = mac
+			d, n, err := w.translate(v)
+			if err != nil {
+				span.SetStatus(codes.Error, err.Error())
+
+				return nil, nil, err
+			}
+			span.SetAttributes(d.EncodeToAttributes()...)
```
Comment (nshalman) [inline backend/file/file.go:126]: This seems kind of wrong to me inside the loop. What are we trying to do here?

**Review by nshalman — APPROVED** [review]
This seems fine to me.
I didn't read the tests very closely, though.

**Comment by jacobweinstock** [comment]
> This seems fine to me. I didn't read the tests very closely, though.

Thank you!

**Pull request merged**